### DISCUSSION TITLE
How do you rate konva？

### DISCUSSION THREAD

**Discussion by leegean** [fabricjs/fabric.js#8768] [General]
I am currently looking for canvas library, and the candidates are mainly [konva](https://konvajs.org) and fabric, which basically meet the requirements in terms of the functions they provide. Could you tell me something else about their differences?

**Comment by ShaMan123** [comment]
I remember when I looked into both konva seemed too strict in terms of subclassing and customization but that was a long time ago.

**Comment by iocoker** [comment]
The dealbreaker for us was that Konva doesn't export to SVG. Aside that, it's an equally great library; just do a feature by feature comparison to see if either supports all the functionalities that are a must for your project.

**Reply by ShaMan123** [reply]
I wasn't aware of that. SVG is not my thing.
BTW I am not sure it is a dealbreaker any more since there is an svg renderer for canvas 2d api. I put it in the readme

**Reply by iocoker** [reply]
We use it for web to print. In W2P, you most likely need PDF and that PDF has to be in vector, not raster. Generating PDF from the JSON will mean reconstructing the objects on a PDF canvas, where object interpretations vary. So, SVG is a bridge between Fabric and the PDF space. It's the same thing if you want to take it into IDML, EPS or any other platform, they all understand SVG.

The JSON fabricjs/Konva generates are unique to fabric/konva. SVG is like a standard pipeline to take the content anywhere while maintaining the exact WYSIWYG representation in vector. Image is another, but then, an image is flat.

For us, an svg renderer for canvas 2d api won't work, as we sometime need to add additional properties to the SVG tags, like spot colors, overprint etc.
